syntax = "proto3";

service ReportService {
    rpc ReportGenerate (ReportGenerateRequest) returns (ReportGenerateReply) {}
    rpc ReportExport (ReportExportRequest) returns (ReportExportReply) {}
}

message RTransaction {
    string owner_name = 1;
    string transaction_name = 2;
    double amount = 3;
    string date = 4;
}

message Report {
    repeated RTransaction transactions = 1;
    double total_income = 2;
    int32 total_operations = 3;
}

message ReportGenerateRequest {
    repeated RTransaction transactions = 1;
}

message ReportGenerateReply {
    int32 status = 1;
    optional Report report = 2;
}

message ReportExportRequest {
    Report report = 1;
}

message ReportExportReply {
    int32 status = 1;
    optional string report_name = 2;
}
